spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -160,13 +162,25 @@ public void executeTask(String taskName, Map<String, String> runtimeProperties,
 		}
 	}
 
+	private Map<String, String> extractAppProperties(TaskDefinition taskDefinition, Map<String, String> taskDeploymentProperties) {
+		final String prefix = "app." + taskDefinition.getName() + ".";
```

Comment:
Same here, we should use `getRegisteredAppName()` since we want the name of the app and not the task definition